src/server.py

```diff
@@ -33,14 +33,31 @@ def play(self):
     def stop(self):
         self.pipeline.send_event(Gst.Event.new_eos())
 
+    @property
+    def volume(self):
+        if "volume" in self.metadata:
+            return self.metadata["volume"]
+        return None
+
     @property
     def sound_location(self):
         return self.metadata["sound-file"]
 
     def _build_pipeline(self):
-        spipeline = ("filesrc location=\"%s\" ! decodebin ! autoaudiosink" %
-                     self.sound_location)
-        return Gst.parse_launch(spipeline)
+        if self.volume is None:
```

Comment:
Since we are going to potentially add more elements and combinations of them (e.g. volume, pitch shift, volume plus pitch shift), it makes sense to me to assemble the pipeline dynamically. You could either do this by using `Gst.ElementFactory` or just by joining strings:
```py
elements = [
    'filesrc location="{}"'.format(self.sound_location),
    'decodebin',
]
if self.volume is not None:
    elements.append('volume volume={}'.format(self.volume))
elements.append('autoaudiosink')
spipeline = ' ! '.join(elements)
```

Reply:
I'd prefer to assemble the pipeline dynamically, but this is an easier to read code.